collect_app/src/test/java/org/odk/collect/android/instancemanagement/InstanceListItemViewTest.kt

```diff
@@ -28,61 +27,42 @@ class InstanceListItemViewTest {
     }
 
     @Test
-    fun whenInstanceIsInvalid_showsIncompleteChip() {
+    fun `show a chip if the status is STATUS_INVALID`() {
         val binding = FormChooserListItemBinding.inflate(layoutInflater)
         val instance = InstanceFixtures.instance(status = Instance.STATUS_INVALID)
 
         InstanceListItemView.setInstance(binding.root, instance, false)
 
         assertThat(binding.chip.visibility, equalTo(View.VISIBLE))
-        assertThat(binding.chip.text, equalTo(context.getString(string.draft_errors)))
```

Comment:
This isn't quite what I meant. There's currently no assertion that checks that we pass the right `Status` to `ErrorsPill`.

In the original comment (https://github.com/getodk/collect/pull/6000#discussion_r1535606109), my example used a property instead of a "setup" method. The idea there was that in this test (and in `SelectionSummarySheetTest`) you could then make an assertion like:

```kotlin
assertThat(binding.chip.status, equalTo(Status.NO_ERRORS))
```

That let's you set a nice boundary in these tests where you delegate all the responsibility of how `NO_ERRORS` actually looks to `ErrorsPillTest` - from `InstanceListItemView` (and its test) that's something an "external" component that just works.

Reply:
Ah ok it should be improved now.